akka-docs/rst/scala/persistence-schema-evolution.rst

```diff
@@ -0,0 +1,386 @@
+.. _persistence-schema-evolution-scala:
+
+##############################
+Persistence - Schema Evolution
+##############################
+
+When working on long running projects using :ref:`persistence-scala`, or any kind of `Event Sourcing`_ architectures,
+schema evolution becomes one of the more important technical aspects of developing your application since both
+requirements as well as our own understanding of the business domain may (and will) change in time.
+
+In fact, if a project matures to the point where you need to evolve its schema to adapt to changing business
+requirements you can view this as first signs of its success – if you wouldn't need to adapt anything over an apps
+lifecycle that could mean that no-one is really using it actively.
+
+In this chapter we will investigate various schema evolution strategies and techniques from which you can pick and
+choose the ones that match your domain and challange at hand.
+
+.. note::
+  This page proposes a number of possible solutions to the schema evolution problem and explains how some of the
+  utilities Akka provides can be used to achieve this, it is by no means a complete (closed) set of solutions.
+
+  Sometimes based on the capabilities of your serialization formats you may be able to evolve your schema in
+  different ways than outlined in the sections below. If you discover useful patterns or techniques for schema
+  evolution feel free to submit Pull Requests to this page to extend it.
+
+
+Schema evolution in event-sourced systems
+=========================================
+
+In recent years we have observed a tremendous move towards immutable append-only datastores, with event-sourcing being
+the prime technique successfully being used in these settings. For an excellent overview why and how immutable data makes scalability
+and systems design much simpler you may want to read Pat Helland's excellent `Immutability Changes Everything`_ whitepaper.
+
+Since with `Event Sourcing`_ the **events are immutable** and usually never deleted – the way schema evolution is handled in such
+systems differs from how one would go about it in a mutable database setting (e.g. in typical CRUD database applications).
+The system needs to be able to continue to work in the presence of "old" events which were stored under the "old" schema,
+and we want to limit the complexity to the data layer, exposing a consistent view over all of the events of a given type
+to :class:`PersistentActor`` s and persistence queries, which allows these layers to focus on the business problems instead
+handling the different schemas explicitly in the business logic layers.
```

Comment:
This sentence is very very long indeed and I didn't understand it until I read it three times. Also, is "data layer" the term you guys are using for the individual user-implementations of `PersistentActor`? I would probably call that the "business layer" or the "business logic layer". 

Suggestion:

>The system needs to be able to continue to work in the presence of "old" events which were stored under the "old" schema. We also want to limit complexity in the business logic layer, exposing a consistent view over all of the events of a given type to `PersistentActor`s and persistence queries. This allows the business logic layer to focus on solving business problems instead of having to explicitly deal with different schemas.